src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Loader } from '../components/Loader';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+import { ErrorType } from '../types/ErrorType';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { filteringPeople } from '../utils/filteringPeople';
+import { sortingPeople } from '../utils/sortingPeople';
+import { getSearchWith } from '../utils/searchHelper';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState<ErrorType>(ErrorType.NONE);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const centuries = searchParams.getAll('centuries') || '';
+  const sex = searchParams.get('sex') || '';
+
+  const showError = (error: ErrorType) => {
+    setIsError(error);
+    setTimeout(() => setIsError(ErrorType.NONE), 3000);
```

Comment:
You should clear all timers on the component unmount
https://bobbyhadz.com/blog/react-cleartimeout